examples/adjoints/adjoints_ex6/adjoints_ex6.C

```diff
@@ -0,0 +1,526 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2015 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// <h1>Adjoints Example 6 - Poisson Equation in square domain with AdjointRefinementErrorEstimator</h1>
+// \author Vikram Garg
+// \date 2017
+//
+// This example solves the Poisson equation, whose solution displays a sharp layer,
+// with QoI based adjoint error estimation and adaptive mesh refinement. The exact
+// solution is . This example also illustrates the use of the adjoint Dirichlet boundary
+// condition capability, necesary for handling flux QoIs. Since we use the adjoint capabilities
+// of libMesh in this example, we use the DiffSystem framework. This file (adjoints_ex6.C)
```

Comment:
This seems misleading - FEMSystem is the only thing with *all* our adjoints support, but we can do adjoint solves with any ImplicitSystem subclass, and this example doesn't actually call any unsupported-by-default code paths, does it?